rust-runtime/aws-smithy-runtime-api/src/client/http/request.rs

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::box_error::BoxError;
+use crate::client::interceptors::context::Error;
+use aws_smithy_http::body::SdkBody;
+use http as http0;
+use std::borrow::Cow;
+use std::fmt::{Debug, Formatter};
+use std::str::FromStr;
+
+/// An HTTP Request Type
+pub struct Request<B = SdkBody> {
```

Comment:
Do we want the body to be generic, even?

Reply:
I think so—really no reason _not_ to have it be generic.